storage/mysql/root_store_test.go

```diff
@@ -0,0 +1,324 @@
+	t.Helper()
+	certDER, err := base64.StdEncoding.DecodeString(b64CertDER)
+	if err != nil {
+		t.Fatalf("Unexpected error while preparing testdata: %s", err)
+	}
+	cert, err := x509.ParseCertificate(certDER)
+	if err != nil {
+		t.Fatalf("Unexpected error while preparing testdata: %s", err)
+	}
+	return cert
+}
+
+func mustBytes32(in []byte, t *testing.T) [32]byte {
+	t.Helper()
+	if len(in) != 32 {
+		t.Fatalf("mustBytes32 expects input of length 32, got %d", len(in))
+	}
+	var out [32]byte
+	copy(out[:], in)
+	return out
+}
+
+func mustHexCode32(s string, t *testing.T) [32]byte {
+	t.Helper()
+	decoded, err := hex.DecodeString(s)
+	if err != nil {
+		t.Fatalf("Unexpected error while preparing testdata: %s", err)
+	}
+	return mustBytes32(decoded, t)
+}
+
+type rootEntry struct {
+	RootID    []byte
+	SHA256DER []byte
+}
+
+type setEntry struct {
+	RootSetID []byte
+	RootID    []byte
+}
+
+type obEntry struct {
+	LogName    string
+	RootSetID  []byte
+	ReceivedAt time.Time
+}
+
+func checkContents(ctx context.Context, t *testing.T, want []rootEntry, wantSets []setEntry, wantObservations []obEntry) {
+	t.Helper()
+
+	tx, err := testDB.BeginTx(ctx, nil /* opts */)
+	if err != nil {
+		t.Fatalf("failed to create transaction: %v", err)
+	}
+	defer tx.Commit()
+
+	// Roots
+	rows, err := tx.QueryContext(ctx, "SELECT Id, DER FROM Roots;")
+	if err != nil {
+		t.Fatalf("failed to query rows: %v", err)
+	}
+	defer rows.Close()
+
+	var got []rootEntry
+	for rows.Next() {
+		var e rootEntry
+		if err := rows.Scan(&e.RootID, &e.SHA256DER); err != nil {
+			t.Fatalf("failed to scan row: %v", err)
+		}
+		got = append(got, e)
+	}
+	if err := rows.Err(); err != nil {
+		t.Errorf("root table iteration failed: %v", err)
+	}
+	if diff := cmp.Diff(got, want); diff != "" {
+		t.Errorf("root table: diff (-got +want)\n%s", diff)
+	}
+
+	// RootSets
+	setrows, err := tx.QueryContext(ctx, "SELECT RootSetID, RootID FROM RootSets;")
+	if err != nil {
+		t.Fatalf("failed to query rows: %v", err)
+	}
+	defer setrows.Close()
+
+	var gotS []setEntry
+	for setrows.Next() {
+		var e setEntry
+		if err := setrows.Scan(&e.RootSetID, &e.RootID); err != nil {
+			t.Fatalf("failed to scan row: %v", err)
+		}
+		gotS = append(gotS, e)
+	}
+	if err := rows.Err(); err != nil {
+		t.Errorf("rootset table iteration failed: %v", err)
+	}
+	if diff := cmp.Diff(gotS, wantSets); diff != "" {
+		t.Errorf("rootset table: diff (-got +want)\n%s", diff)
+	}
+
+	// RootSetObservations
+	obrows, err := tx.QueryContext(ctx, "SELECT LogName, RootSetID, ReceivedAt FROM RootSetObservations;")
+	if err != nil {
+		t.Fatalf("failed to query rows: %v", err)
+	}
+	defer obrows.Close()
+
+	var gotO []obEntry
+	for obrows.Next() {
+		var e obEntry
+		if err := obrows.Scan(&e.LogName, &e.RootSetID, &e.ReceivedAt); err != nil {
+			t.Fatalf("failed to scan row: %v", err)
+		}
+		gotO = append(gotO, e)
+	}
+	if err := obrows.Err(); err != nil {
+		t.Errorf("rootsetobservation table iteration failed: %v", err)
```

Comment:
nit: The table name has an "s" on the end.

```suggestion
		t.Errorf("RootSetObservations table iteration failed: %v", err)
```

Reply:
Also applies to the next error message.